File: PLP_python/Aula03_ListaDeExercicios(15)/Q05/Q05_main.py
from Q05_dog import Dog
from Q05_cat import Cat


def make_sounds(animals):
    """
    Receives a list of Animal objects and calls the make_sound method of each one.

    Args:
        animals (list): List of Animal objects.
    """
    for animal in animals:
        animal.make_sound()


def main():
    dog = Dog()
    cat = Cat()
    animals = [dog, cat]

    make_sounds(animals)


if __name__ == "__main__":
    main()

File: PLP_Java/src/Aula03_ListaDeExercicio/Q05/Q05_Main.java
package Aula03_ListaDeExercicio.Q05;

public class Q05_Main {
    /**
     * Receives a list of Animal objects and calls the makeSound method of each one.
     */
    public static void makeSounds(Q05_Animal[] animals) {
        for (Q05_Animal animal : animals) {
            animal.makeSound();
        }
    }

    public static void main(String[] args) {
        Q05_Animal dog = new Q05_Dog();
        Q05_Animal cat = new Q05_Cat();
        Q05_Animal[] animals = {dog, cat};

        makeSounds(animals);
    }
}
